state/upgrades.go

```diff
@@ -2086,3 +2086,18 @@ func EnsureApplicationDeviceConstraints(pool *StatePool) error {
 	}
 	return nil
 }
+
+// RemoveInstanceCharmProfileDataCollection removes the
+// instanceCharmProfileData collection on upgrade.
+func RemoveInstanceCharmProfileDataCollection(pool *StatePool) error {
```

Comment:
This is such a good idea, I didn't think about this... this should ensure that no existing upgrades cause erroneous errors.